os/pm/pm_changestate.c

```diff
@@ -114,10 +114,12 @@ static void pm_timer(int domain)
 
 	if (pdom->state < PM_SLEEP && !pdom->stay[pdom->state] && pmtick[pdom->state]) {
 		int delay = pmtick[pdom->state] + pdom->btime - clock_systimer();
```

Comment:
Delay is logically calculated as: -
Total waiting time in current state before moving to next state - (time already elapsed in current state)
The latter is calculated as (current system time - system time at which current state was started)

If this delay is 0, then that means that the time duration for waiting in current state has already elapsed.
This is possible if: -
-> The value of `TIME_SLICE_TICKS * CONFIG_PM_<STATE>ENTER_COUNT` is too less
In this case, I think we should add a comment to the code to inform the developer.
We will have a practical lower limit to the config mentioned above, which will be determined as we go ahead.
OR
-> We have already stayed in this state for a long time, which is not expected because pm_timer() is called as soon as the state is changed or a state change fails and the timer for the current state is restarted.


My reason for writing this is that when delay<=0, you are manually changing it to delay=1
This starts a watchdog timer for 1 system tick only.
But we should be more concerned about how we got delay<=0, and such cases should be avoided in the first place.

Reply:
@edwakuwaku what's your opinion on this?

Reply:
Regarding the first point, I agree with you that we should add in some kind of lower bound for the state drop counter.
For the second point, could you please share the scenario that might cause staying a state for a long time? As far as I understand, this should not happen according to the PM driver realization.
This change is to prevent giving negative input to wdog timer (referenced from NUTTX), I'm not sure this is added because they encountered problem similar as to your concerned, or just a basic protection. Anyway, I am on the same page with you regarding this concern, and due to short timespan, I think we should proceed to conduct test on application first, and get back to review this point if we encounter any problem related to this logic.

Reply:
For the second point, as I mentioned, it is not expected.
However, developers face errors only when something unexpected happens. We should have safeguards in place for such unexpected cases.
Please add a debug statement here (pmdbg) to print log if delay<=0 is observed
Then, we will continue to manually reset delay to 1 to pass to wdog function as you have implemented.

If we face such cases and they cannot be resolved by increasing wait duration, we will check what to do at that time.